Modules/Loadable/SubjectHierarchy/Widgets/qSlicerSubjectHierarchyViewContextMenuPlugin.h

```diff
@@ -50,6 +50,15 @@ class Q_SLICER_MODULE_SUBJECTHIERARCHY_WIDGETS_EXPORT qSlicerSubjectHierarchyVie
 
 public:
 
+  enum {
+       ID_CTBone,
+       ID_CTAir,
+       ID_PET,
+       ID_CTAbdomen,
+       ID_CTBrain,
+       ID_CTLung,
+       ID_DTI,
+  };
```

Comment:
Great workflow and got several aspects realized,  but unfortunately, I am unable to call 

volumeModulePointer = d->ModuleManager->module("Volumes");

from  qSlicerSubjectHierarchyViewContextMenuPluginPrivate::init() because the module "Volume" seems not yet to have loaded at the time when init() is executed. Any ideas or can I somehow directly reference the module and the connected logic()?

Reply:
It is possible for modules to express dependencies, but it's not a common requirement and probably better to avoid it.
https://github.com/Slicer/Slicer/blob/0094e9a35bd266cc8b0e677858dabce797c9928f/Base/QTCore/qSlicerAbstractCoreModule.h#L154-L160

For this reason and others I'd still suggest moving the preset information into the display node instead of having the subject hierarchy plugin access the volume module (in general we'd want all mrml operations to work even if modules and especially module guis are not instantiated.

Reply:
> modules to express dependencies

Thanks, bút I was not able to define dependencies in qSlicerSubjectHierarchyViewContextMenuPlugin, probably because it is not a loadable module.

Reply:
I just realized that you modified the common base class qSlicerSubjectHierarchyViewContextMenuPlugin.cxx. That is only for adding generic view menu actions. Volumes related actions should be added to qSlicerSubjectHierarchyVolumesPlugin.cxx in viewContextMenuActions() method, similarly how it is done for markups in qSlicerSubjectHierarchyMarkupsPlugin::viewContextMenuActions().